core/src/main/java/org/apache/carbondata/core/datastore/impl/FileFactory.java

```diff
@@ -43,11 +45,19 @@
    */
   private static final LogService LOGGER =
       LogServiceFactory.getLogService(FileFactory.class.getName());
-  private static Configuration configuration = null;
 
-  static {
-    configuration = new Configuration();
+  public static Configuration getConfiguration() {
```

Comment:
1) Please segregate all cofig handling to CarbonConfiguration. You can add a util function as getConfFromSession
2) FileFactory.getDataInputStream already accepts the config parameter. I suggest to unify the callers of this method also through CarbonConfiguration

Reply:
removed config parameters from FileFactory